engine/src/test/java/com/redhat/cloud/notifications/processors/email/EmailSubscriptionTypeProcessorWithMigratedTemplateTest.java

```diff
@@ -0,0 +1,232 @@
+
+    @InjectMock
+    @RestClient
+    ITUserService itUserService;
+
+    @InjectSpy
+    EmailTemplateFactory emailTemplateFactory;
+
+    @InjectMock
+    WebhookTypeProcessor webhookSender;
+
+    @InjectSpy
+    TemplateRepository templateRepository;
+
+    @Inject
+    FeatureFlipper featureFlipper;
+
+    String commonTest() {
+        mockGetUsers(8);
+
+        final String tenant = "instant-email-tenant";
+        final String[] usernames = {"username-1"};
+        String bundle = "rhel";
+        String application = "policies";
+
+        for (String username : usernames) {
+            subscribe(DEFAULT_ORG_ID, username, bundle, application);
+        }
+
+        final List<String> bodyRequests = new ArrayList<>();
+
+        doAnswer(invocation -> {
+            Object[] args = invocation.getArguments();
+            bodyRequests.add(String.valueOf(args[3])); // The fourth argument is email Payload
+            return null;
+        }).when(webhookSender)
+            .doHttpRequest(any(), any(), any(), any(), anyString(), anyString(), anyBoolean());
+
+        Action emailActionMessage = TestHelpers.createPoliciesAction(tenant, bundle, application, "My test machine");
+        List<EventType> events =  getEventTypes();
+
+        Event event = new Event();
+        event.setEventType(events.get(0));
+        event.setAction(emailActionMessage);
+
+        EmailSubscriptionProperties properties = new EmailSubscriptionProperties();
+
+        Endpoint ep = new Endpoint();
+        ep.setType(EndpointType.EMAIL_SUBSCRIPTION);
+        ep.setName("positive feeling");
+        ep.setDescription("needle in the haystack");
+        ep.setEnabled(true);
+        ep.setProperties(properties);
+        try {
+            statelessSessionFactory.withSession(statelessSession -> {
+                emailProcessor.process(event, List.of(ep));
+            });
+
+            assertEquals(1, bodyRequests.size());
+            JsonObject email = new JsonObject(bodyRequests.get(0));
+            String bodyRequest = email.getJsonArray("emails").getJsonObject(0).getString("body");
+
+            assertTrue(bodyRequest.contains(TestHelpers.policyId1), "Body should contain policy id" + TestHelpers.policyId1);
+            assertTrue(bodyRequest.contains(TestHelpers.policyName1), "Body should contain policy name" + TestHelpers.policyName1);
+
+            assertTrue(bodyRequest.contains(TestHelpers.policyId2), "Body should contain policy id" + TestHelpers.policyId2);
+            assertTrue(bodyRequest.contains(TestHelpers.policyName2), "Body should contain policy name" + TestHelpers.policyName2);
+
+            // Display name
+            assertTrue(bodyRequest.contains("My test machine"), "Body should contain the display_name");
+
+            assertTrue(bodyRequest.contains(TestHelpers.HCC_LOGO_TARGET));
+
+            return bodyRequest;
+        } catch (Exception e) {
+            e.printStackTrace();
+            fail(e);
+        }
```

Comment:
Is that necessary? If the method `throws Exception`, wouldn't the test fail in a similar way?